src/NuGet.Core/NuGet.Commands/VerifyCommand/VerifyCommandRunner.cs

```diff
@@ -58,12 +61,31 @@ public async Task<int> ExecuteCommandAsync(VerifyArgs verifyArgs)
                         _defaultFingerprintAlgorithm)).ToList();
 
                 var verifierSettings = SignedPackageVerifierSettings.GetVerifyCommandDefaultPolicy();
+
+                // nuget.config >> trustedSigners section read
+                IReadOnlyList<TrustedSignerAllowListEntry> trustedSignerAllowList = TrustedSignersProvider.GetAllowListEntries(verifyArgs.Settings, verifyArgs.Logger);
+
                 var verificationProviders = new List<ISignatureVerificationProvider>()
                 {
-                    new IntegrityVerificationProvider(),
-                    new SignatureTrustAndValidityVerificationProvider()
+                    new IntegrityVerificationProvider()
                 };
 
+                // trustedSigners section >> Owners are considered here.
+                verificationProviders.Add(
+                      new AllowListVerificationProvider(
+                          trustedSignerAllowList,
+                          requireNonEmptyAllowList: clientPolicyContext.Policy == SignatureValidationMode.Require,
+                          emptyListErrorMessage: Strings.Error_NoClientAllowList,
+                          noMatchErrorMessage: Strings.Error_NoMatchingClientCertificate));
+
+                IEnumerable<KeyValuePair<string, HashAlgorithmName>> trustedSignerAllowUntrustedRootList = trustedSignerAllowList?
+                    .Where(c => c.AllowUntrustedRoot)
+                    .Select(c => new KeyValuePair<string, HashAlgorithmName>(c.Fingerprint, c.FingerprintAlgorithm));
+
+                // trustedSigners section >> allowUntrustedRoot set true are considered here.
+                verificationProviders.Add(new SignatureTrustAndValidityVerificationProvider(trustedSignerAllowUntrustedRootList));
+
+                // List of values passed through --certificate-fingerprint option are considered here.
```

Comment:
// For verify action repository signing allow list not considered.

https://github.com/NuGet/NuGet.Client/blob/b8fe2a0f84a2a1a7f4efe2dcaaa7876bef8f4627/src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs#L1019-L1029